Python/pytime.c

```diff
@@ -427,6 +427,15 @@ _PyTime_FromSeconds(int seconds)
 }
 
 
+_PyTime_t
+_PyTime_FromSecondsDouble(double seconds)
+{
+    _PyTime_t t = (_PyTime_t)(seconds * SEC_TO_NS);
```

Comment:
Rounding is undefined here. You should reuse pytime_from_double() with SEC_TO_NS, and use the right rounding mode in your code. time.clock_settime() uses _PyTime_ROUND_FLOOR rounding mode.

Reply:
@vstinner 
Fixed.